model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/functional_batch_norm.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from torch import nn
+import torch.nn.functional as F
+import copy
+
+from model_compression_toolkit.core.common.graph.graph_matchers import NodeOperationMatcher
+from model_compression_toolkit.core import common
+from model_compression_toolkit.core.common import BaseNode, Graph, FrameworkInfo
+from model_compression_toolkit.core.pytorch.constants import *
+
+
+def functional_batch_norm_matcher() -> NodeOperationMatcher:
+    """
+    Function generates matchers for matching:
+    functional.batch_norm.
+
+    Returns:
+        Matcher for batch_norm node.
+    """
+    bn_node = NodeOperationMatcher(F.batch_norm)
+    return bn_node
+
+
+class FunctionalBatchNorm(common.BaseSubstitution):
+    """
+    Replace functional batch_norm with BatchNorm2d.
+    """
+
+    def __init__(self, ):
+        """
+        Matches: functional batch_norm
+        """
+        bn_node = NodeOperationMatcher(F.batch_norm)
+        super().__init__(matcher_instance=bn_node)
+
+    def get_attributes_from_inputs(self, graph: Graph, node: BaseNode) -> dict:
+        input_nodes = graph.get_prev_nodes(node)
+        if len(input_nodes) == 5:
```

Comment:
make sure len(input_nodes) == 4 isn't an option

Reply:
there's and error if "else"

Reply:
I saw that. I meant that it's not an option in torch. for example if it is craeted like this:
`torch.nn.functional.batch_norm(input, running_mean, running_var, weight=weight, bias=None`
the bias is still created

Reply:
if weight\bias = None is doesn't create the input node